### PR TITLE
DS-5171 As a SM I want to keep Revisions for Data Policy

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/DataPolicyAccessControlHandler.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * @see \Drupal\gdpr_consent\Entity\DataPolicy.
+ */
+class DataPolicyAccessControlHandler extends EntityAccessControlHandler {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function access(EntityInterface $entity, $operation, AccountInterface $account = NULL, $return_as_object = FALSE) {
+    $access = parent::access($entity, $operation, $account, $return_as_object);
+    $x = 1;
```
Comment (bramtenhove) [inline src/DataPolicyAccessControlHandler.php]: @OH17L this seems out of place. Perhaps remove it?

File: src/DataPolicyTranslationHandler.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Drupal\gdpr_consent;
+
+use Drupal\content_translation\ContentTranslationHandler;
+
+/**
+ * Defines the translation handler for data_policy.
+ */
+class DataPolicyTranslationHandler extends ContentTranslationHandler {
```
Comment (bramtenhove) [inline src/DataPolicyTranslationHandler.php]: @OH17L this class does nothing. Shouldn't we just point to the normal class? E.g. `ContentTranslationHandler`?

File: src/Entity/DataPolicy.php [hunk 3]
```diff
@@ -16,41 +17,95 @@
  * @ContentEntityType(
  *   id = "data_policy",
  *   label = @Translation("Data policy"),
- *   label_collection = @Translation("Data policy consents"),
  *   handlers = {
- *     "list_builder" = "Drupal\gdpr_consent\DataPolicyListBuilder",
+ *     "storage" = "Drupal\gdpr_consent\DataPolicyStorage",
+ *     "view_builder" = "Drupal\Core\Entity\EntityViewBuilder",
+ *     "views_data" = "Drupal\gdpr_consent\Entity\DataPolicyViewsData",
+ *     "translation" = "Drupal\gdpr_consent\DataPolicyTranslationHandler",
```
Comment (bramtenhove) [inline src/Entity/DataPolicy.php]: Perhaps use `Drupal\content_translation\ContentTranslationHandler` as the `DataPolicyTranslationHandler` is empty?

File: src/Form/DataPolicySettingsForm.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  public function getFormId() {
+    return 'gdpr_consent_data_policy_settings';
+  }
+
+  /**
+   * Form submission handler.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
```
Comment (bramtenhove) [inline src/Form/DataPolicySettingsForm.php]: @OH17L, just wondering. But could we not simply use `{@inheritdoc}` here (and also below)?

File: src/DataPolicyAccessControlHandler.php (hunk 1 above)
Comment (OH17L) [reply src/DataPolicyAccessControlHandler.php]: @bramtenhove, yes, I will delete it.

File: src/DataPolicyTranslationHandler.php (hunk 2 above)
Comment (OH17L) [reply src/DataPolicyTranslationHandler.php]: @bramtenhove, I will check it.

File: src/Entity/DataPolicy.php (hunk 3 above)
Comment (OH17L) [reply src/Entity/DataPolicy.php]: @bramtenhove, I will check it.

File: src/Form/DataPolicySettingsForm.php (hunk 4 above)
Comment (OH17L) [reply src/Form/DataPolicySettingsForm.php]: @bramtenhove you are right. I will change it.

**Comment by bramtenhove** [comment]
@OH17L I noticed that when I create a new revision it also copies the revision comment I wrote in the previous one.
I think we should not autofill the revision comment from the previous revision. So just leave it empty.

File: src/Entity/DataPolicy.php (hunk 3 above)
Comment (bramtenhove) [reply src/Entity/DataPolicy.php]: I've been testing translations. After removing this I got errors when enabling content translation for Data Policy.
Maybe put this back?

**Comment by bramtenhove** [comment]
@OH17L, can you explain how I can translate the data policy? When I reverted back to the previous commit (with the custom translation handler class) I am able to create a data policy in a specific language. However I cannot find a button or anything to translate the data policy. Any clue?

**Comment by bramtenhove** [comment]
@OH17L, I will merge this PR. We can further improve translations in a different story.

**Pull request merged**